exercise_solutions.tex

```diff
@@ -1467,6 +1467,62 @@ \subsection*{Solution to \cref{ex:is-conn-trunc-functor}}
   &= (t \circ \tprojf k)(a) \tag{since $g$ is a section of $\lam{s} s\circ f$}\\
   & \equiv t(\tproj ka).
 \end{align*}
+\subsection*{Solution to \cref{ex:categorical-connectivity}}
+
+Let us denote by $\mathsf{isConn}(A)$ the connectivity of type $A$. It is known that
+\begin{displaymath}
+ \mathsf{isConn}(A) \simeq \trunc {} A \times \prd{x,y:A} \trunc {} {\id[A]xy}
+\end{displaymath}
+(see \cref{ex:connectivity-inductively}).
+\begin{enumerate}
+ \item Suppose a type $A$ is connected. The goal $\prd{B,C:\type} \isequiv(e_{A,B,C})$ is a mere proposition, so we can assume $a:A$ and $\prd{x,y:A} \trunc {} {x = y}$.
+
+       Let $B,C$ be types. We want to construct $h:(A\to B+C) \to (A\to B)+(A\to C)$ such that $h \circ e = \idfunc$ and $e \circ h = \idfunc$.
+       Let $f:A \to B+C$ be an arbitrary function. We have case analysis on $f(a):B + C$.\\
+       (a) if $f(a) \equiv \inl(u)$ for some $u:B$:
+       We want a function $g:A\to B$ such that $f(x) = \inl(g(x))$ for every $x:A$. Let us define $k:B+C\to B$ by
+       \begin{align*}
+	k(\inl(b)) &\defeq b;\\
+	k(\inr(c)) &\defeq f(a) \enspace,
+       \end{align*}
+       and $g \defeq k \circ f$. Then we have $\inl(g(x)) = \inl(k(f(x))) = f(x)$ for every $x:A$ (by case analysis on $f(x)$ and $\trunc {} {\id[A]{a}{x}}$).
+       With this $g$, we define $h(f) \defeq \inl(g)$. Then we have
+       \begin{displaymath}
+	e_{A,B,C}(h(f)) = e_{A,B,C}(\inl(g)) = \lam{x} \inl(g(x)) = f
+       \end{displaymath}
+       and
+       \begin{displaymath}
+	h(e_{A,B,C}(\inl(v))) = h(\lam{x} \inl(v(x))) = \lam{x} k(\inl(v(x))) = v
+       \end{displaymath}
+       for any $v: A\to B$.\\
+       (b) if $f(a) \equiv \inr(u)$ for some $u:C$: Similar to (a).\\
+ \item (Categorial connectivity to $\mathsf{isConn}$ implies $\LEM{}$):
+       Assume for every type $A$,
+       $(\prd{B,C:\type}\isequiv(e_{A,B,C})) \to \mathsf{isConn}(A)$.
+       Assuming $\neg \neg P$ for a mere proposition $P$, we want to prove $P$.
+       Let us define $A \defeq \susp P$.
+       Since $P \simeq (\id[\susp P]{\north}{\south}) \simeq \trunc {} {\id[\susp P]{\north}{\south}}$ (by \cref{prop:trunc_of_prop_is_set}
+       and since $P$ is mere) and $\mathsf{isConn}(\susp P)$ implies $\trunc {} {\id[\susp P]{\north}{\south}}$ (by \cref{ex:connectivity-inductively}),
+       it suffices to prove $\prd{B,C:\type}\isequiv(e_{\susp P,B,C})$. The proof is done by case analysis on $(f(\north), f(\south))$ for $f:\susp P\to B+C$.
+
+       ($\LEM{}$ implies categorial connectivity to $\mathsf{isConn}$):
+       Assume $\LEM{}$, $A:\type$ and $\prd{B,C:\type} \isequiv(e_{A,B,C})$.
+       We need to prove $\mathsf{isConn}(A)$.
+       By $\LEM{}$ We have $\trunc {} A$ because if $\neg \trunc {} A$ we have $A \simeq 0$ and $1 + 1 \simeq 1$, which is not the case.
```

Comment:
"We" should be "we".  Also this reads a bit run-on, I would put a comma before "because".